src/care/host_device_ptr.h

```diff
@@ -206,11 +222,19 @@ namespace care {
 #if !CARE_LEGACY_COMPATIBILITY_MODE
       const
 #endif
-      { return MA::operator[](i); }
+      {
+#if !defined(CARE_DEVICE_COMPILE) && defined(CARE_ENABLE_BOUNDS_CHECKING)
+         boundsCheck(i);
+#endif
+         return MA::operator[](i);
+      }
 
 #if CARE_LEGACY_COMPATIBILITY_MODE
      template<typename Idx>
      inline CARE_HOST_DEVICE T& operator[](const Idx i) const {
+#if !defined(CARE_DEVICE_COMPILE) && defined(CARE_ENABLE_BOUNDS_CHECKING)
+        boundsCheck(i);
```

Comment:
Does this need to be added to the pick() function separately?

Reply:
Done.